File: fluent-bot-lib/src/main/java/org/khasanof/core/utils/AnnotationUtils.java
package org.khasanof.core.utils;

import java.lang.annotation.Annotation;
import java.lang.reflect.Method;
import java.util.Arrays;
import java.util.Objects;

/**
 * @author Nurislom
 * @see org.khasanof.core.utils
 * @since 12.07.2023 23:41
 */
public abstract class AnnotationUtils {

    public static boolean hasAnnotation(Method method, Class<? extends Annotation> annotation, boolean isSuper) {
        if (Objects.nonNull(method)) {
            if (isSuper) {
                return Arrays.stream(method.getAnnotations())
                        .anyMatch(any -> any.annotationType()
                                .isAnnotationPresent(annotation));
            }
            return method.isAnnotationPresent(annotation);
        }
        return false;
    }

}

File: khasanof-common-util/src/main/java/org/khasanof/annotation/AnnotationUtils.kt
package org.khasanof.annotation

import org.khasanof.base.BaseUtils
import java.lang.reflect.Method
import java.util.Arrays
import java.util.Objects

fun hasAnnotation(method: Method, annotation: Class<out Annotation>, isSuper: Boolean): Boolean {
    BaseUtils.checkArgsIsNonNull(method, annotation, isSuper)
    if (isSuper) {
        return Arrays.stream(method.annotations)
            .anyMatch { any ->
                Objects.equals(any.annotationClass, annotation)
            }
    }
    return method.isAnnotationPresent(annotation)
}
